### PR TITLE
FIX: Caret moves to a wrong position when uploading an image via toolbar

### PR DESCRIPTION
**Pull request by AndrewPrigorshnev**
When uploading an image, we change the uploading placeholder several times. Every time, we correct the position of the cursor after replacing. But we schedule repositioning of cursor to the `afterRender` queue in Ember Run Loop. As a result, sometimes we replace the placeholder several times but correct the cursor position only once at the end. 

It just cannot work correctly with scheduling, we'll always be dealing with cumulative error. Removing scheduling fixes the problem.

Sadly, I cannot make the test work, I skipped it for now, going to give it another try later.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/d-editor.js [hunk 1]
```diff
@@ -477,7 +477,7 @@ export default Component.extend(TextareaTextManipulation, {
       key: "#",
       afterComplete: (value) => {
         this.set("value", value);
-        return this._focusTextArea();
+        return schedule("afterRender", this, this._focusTextArea);
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/app/components/d-editor.js:480]: Do you need the `return` here?

File: app/assets/javascripts/discourse/tests/acceptance/composer-uploads-uppy-test.js [hunk 2]
```diff
@@ -310,6 +310,30 @@ acceptance("Uppy Composer Attachment - Upload Placeholder", function (needs) {
     const image = createFile("avatar.png");
     appEvents.trigger("composer:add-files", image);
   });
+
+  skip("should place cursor properly after inserting a placeholder", async function (assert) {
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/tests/acceptance/composer-uploads-uppy-test.js:314]: Why add a test to `skip` it?

**Comment by ZogStriP** [comment]
> Sadly, I cannot make the test work, I skipped it for now, going to give it another try later.

Maybe @cvx or @jjaffeux can help you with that?

**Comment by AndrewPrigorshnev** [comment]
> Maybe @CvX or @jjaffeux can help you with that?

It's not like there is a roadblock that is too hard to deal with. It just needs a bit more time to play with it, when I'm testing it I see that the bug is clearly solved but something unusual is happening in the test environment.

I wanted to merge it to dogfood it a bit and then backport to the release. What if I deal with the test early next week, and ping @CvX or @jjaffeux in case it takes too much time?

File: app/assets/javascripts/discourse/app/components/d-editor.js (hunk 1 above)
Comment (AndrewPrigorshnev) [reply app/assets/javascripts/discourse/app/components/d-editor.js]: We don't need it, you're right.

**Review by ZogStriP — APPROVED** [review]

**Comment by AndrewPrigorshnev** [comment]
Turned out that this fixes the bug only when running under Ember CLI. The PR for the legacy environment is upcoming.

**Pull request merged**